### ISSUE TITLE
Automate the process of voting for TSC members.

### ISSUE THREAD

**Issue by KhudaDad414** [asyncapi/.github#102]
### Reason/Context
Now that we have a list of TSC members, I think it's time to create a voting process for TSC members.
### Description
We can have a new type of issue called `poll` on the `community` repo.
#### How could it be implemented/designed?
1) having the bot do the following jobs:
      * When a poll issue is created it should notify the TSC members by email or mentioning them on the issue.
      * Notify TSC members that haven't voted before the poll expires.
      * After the poll expires, the bot should count the votes and create a comment announcing the results.
2) having an issue template for polls would be useful since it will make it easier for a bot to read it and do its thing.
#### How can a TSC member vote?
we can assign emojis to each option and they can vote by reacting to the issue or they can comment their vote.
#### What about a voting system on the website?
I don't exactly know how that would work. since the website is static and having an authentication system on the website for TSC members only for voting seems a little extra? but it will be fancier though 😃 
Would love to know what others think.
cc: @derberg @fmvilas @alequetzalli

**Comment by fmvilas** [comment]
I'm all in for automation 😄 The only thing that I don't see viable or cool is putting it on the website for TSC members only. Explore the web because there may be some bots already out there 🤔 

Also, another suggestion from my side is making these polls/decisions visible on the website. That would be cool.

**Comment by derberg** [comment]
> When a poll issue is created it should notify the TSC members by email or mentioning them on the issue.

once I merge https://github.com/asyncapi/.github/pull/94 (next week) I will turn upside down all access rights in GH Org and invite all tsc members to be part of AsyncAPI GitHub org and they will have a dedicated group. So we should just have a workflow that once given issue with the specific label is created, the group is automatically assigned 

> on the website for TSC members only.

TSC votes are public and should be exposed to anyone. Since all will be in issues, then in the website you can read github issues and display a list easily. To make things even easier, the list of votes can be like TSC members a list stored in `community` repo and then pushed to `website` the same way we push TSC members list JSON

> issues

I have concerns. The issue is easy but also sucks from a `discussion` point of view. There might be topics that will trigger lots of comments that will be hard to follow. PR is much better, but of course more difficult to create (although is it really a problem?). So maybe instead of `issues` we should go with pull request or discussion in a dedicated TSC voting category? The advantage of PR is:
- approve/reject feature is there so easier to count votes
- tsc members group can be codeowner of directory where we have files with voting topics, and they will automatically be assigned as reviewers

**Comment by KhudaDad414** [comment]
I am not exactly sure I understand the process with PRs. What should I do If I want to create a poll? how can I vote? are we storing the polls in a file or it will be on the PR description. @derberg What is the process that you have in mind for voting in PRs?

**Comment by derberg** [comment]
we would exactly store it in a file, in a repo, in a directory called for example `tsc_votes`. You create a file with the description, maybe pictures, maybe diagrams, whatever, but basically a markdown file with all the details.

vote is either approve or reject the PR.

disadvantage of course is that it ain't that easy to create a PR comparing to Issue creation, but is this really a problem 😄

**Comment by alequetzalli** [comment]
I have been thinking about this and the convo thread so far, but I still feel unsure which option I "subscribe" to. 🤔 😬

I like the idea of opening a Discussion and Poll, because I think that drives more engagement. But I also like how "simple" counting votes via a PR process can be. 

Is there a way to combine both elements? 😀

**Comment by derberg** [comment]
but `poll` mentioned by @KhudaDad414 is "just" a name, right Khuda? Not that there would be an actual poll. It would just be a special issue because of its template and labels. We can also call PR a `poll` if you want 😄 

Using issues is really a big problem, especially when we will have a controversial topic to vote on that trigger discussion. Try to follow this issue https://github.com/asyncapi/spec/issues/618 😆 

Using Discussions is an option, but then we have to code all that stuff ourselves, and the only extra stuff I see it adds is "no need to fork/branch/push/openPR". So yeah, a matter of deciding. Just imagine a topic that requires more explanation, TSC member wants more explanation on certain sentence mentioned in the `poll` -> I think much better experience is PR inline code comments

**Comment by jonaslagoni** [comment]
I think we need to split it up.

I suggest whenever something needs to be decided we create both a discussion and a PR.

PR is for voting **ONLY**, no discussion can take place here. You either reject or accept it. The PR needs to clearly link to where the discussion can take place.

Discussion is for asking questions and general discussion to come to an agreement. After PR is merged the discussion will be closed. This way we don't need to code anything extra for discussions.

That way we get the best of both worlds 🤔

**Comment by derberg** [comment]
any reason why you think Discussions are better than PRs? I'm not yet fully decided on my own
